lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1059,14 +1108,33 @@ public function resetQueryPart(string $queryPartName) : self
      */
     private function getSQLForSelect() : string
     {
-        $query = 'SELECT ' . ($this->sqlParts['distinct'] ? 'DISTINCT ' : '') .
-                  implode(', ', $this->sqlParts['select']);
+        $query = 'SELECT ';
+
+        if ($this->distinct) {
+            $query .= 'DISTINCT ';
+        }
+
+        $query .= implode(', ', $this->select);
```

Comment:
Is there a way that this query will be valid if the select block is empty? We probably should throw a `QueryException` here.

Reply:
Right. Done in ffc65d42e80e87547d56847263fc46c628ccc95b.